### ISSUE TITLE
GoRun issue with passing arguments

### ISSUE THREAD

**Issue by stefanos82** [fatih/vim-go#1834]
Is everything alright with `:GoRun %`?

I have the following code and does not work with the aforementioned command:
```go
package main

import (
	"fmt"
	"os"
	"path/filepath"
	"time"
)

func main() {

	var myDate string
	if len(os.Args) != 2 {
		fmt.Printf("filename: %s\n", filepath.Base(os.Args[0]))
		return
	}

	myDate = os.Args[1]
	date, err := time.Parse("02 January 2006", myDate)
	if err == nil {
		fmt.Println("Full date:", date)
		fmt.Println("Time:", date.Day(), date.Month(), date.Year())
	} else {
		fmt.Println(err)
	}
}
```
As soon as I run it with `:GoRun % "20 July 2008"`, I get `filename: main`.

When I run it with `go run` command I get the proper output:
```bash
$ go run various/dateFormatting/main.go "20 July 2008"
Full date: 2008-07-20 00:00:00 +0000 UTC
Time: 20 July 2008
```

My specs are:
```txt
Vim 8.0.1766
vim-go: the latest (GoUpdateBinaries)
System: GNU / Debian testing 64-bit
Go version: 1.10.1
```

**Comment by Carpetsmoker** [comment]
Note that it's documented as:

    :GoRun[!] [expand]

    [..]

    [expand] is used as file selector

In other words, using arguments is not a documented feature.

Would certainly be useful to add though.

**Comment by bhcleek** [comment]
Fixed via #2842

**Comment by stefanos82** [comment]
Still the same behavior I'm afraid...

Current specs are:
```txt
go version go1.14.2 linux/amd64
VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Mar 24 2020 03:21:40)
Included patches: 1-439
vim-go version: d5bc2aee707f6acbd6dbe07d479950e33fc11a03
vim-go dependencies: fully updated with GoUpdateBinaries (10 minutes ago as of this writing)
```

**Comment by bhcleek** [comment]
The quoting is the problem now; you need to make sure that the date is treated as a single argument by removing the double quotes and escaping the spaces: `GoRun % 20\ July\ 2008`

**Comment by stefanos82** [comment]
Ah I see, yes now it works with escaping the spaces.

This is quite interesting though. Why does it work under terminal via quotes (both single and double), but not the same under `GoRun %`?

**Comment by bhcleek** [comment]
Because the semantics of providing arguments is different in ex mode than in a shell.

**Comment by stefanos82** [comment]
Does this relate to this topic?

https://github.com/vim/vim/issues/459

Seems like they patched it https://github.com/vim/vim/commit/4bfa8af14142e54f509048239f4e8596911f56aa

**Comment by bhcleek** [comment]
No, that's a separate topic.

Looking at this again, I don't think the issue was resolved with #2842: the problem you encountered was only related to trying to use shell semantics for quoting a single argument with spaces instead of escaping the spaces.
